stories/control/legend.stories.tsx

```diff
@@ -51,24 +66,48 @@ type TestProps = LegendProps & { hidden: boolean };
 
 export function WithTwoSeries(props: TestProps) {
   const { hidden, ...otherProps } = props;
+  const [highlightSeries, setHighlightSeries] = useState<boolean[]>([
+    false,
+    false,
+  ]);
+  const updateHiddenSeries = (index: number) => {
+    setHighlightSeries((highlightSeries) => {
+      const result = highlightSeries.map((highlightSerie, i) => {
+        if (i === index) return !highlightSerie;
+        return highlightSerie;
+      });
+      return result;
+    });
+  };
   return (
     <Plot {...DEFAULT_PLOT_CONFIG}>
-      <Legend {...otherProps} />
-
-      <LineSeries data={data1} xAxis="x" yAxis="y" label="Label line series" />
-
+      <Legend
+        {...otherProps}
+        onClick={(index) => () => updateHiddenSeries(index)}
+      />
+      <LineSeries
+        data={data1}
+        lineStyle={{
+          strokeWidth: highlightSeries[0] ? '5' : '',
```

Comment:
Rather than hardcoding the element of the array we could use a callback.

Please allow lineStyle to receive a callback that is called with an object in which one of the properties (and currently the only one) is `id` (the series id).

Reply:
> Rather than hardcoding the element of the array we could use a callback.
could you please explain this

Reply:
something like this: `strokeWidth: ({id} => highlightSeries[id] ? '5' : '')`

Reply:
Thanks, fixed